### PR TITLE
Use KIND instead of DIND & Include Skipped test cases

### LINKED ISSUE #835: Fix e2e test: gang scheduling.
**Is this a BUG REPORT or FEATURE REQUEST?**:

/kind bug

**What happened**:
After the revert of batch job patches #806 , the [gang scheduling](https://github.com/kubernetes-sigs/kube-batch/blob/5e9977d68a13938fbb6d7dda564da0dca1b1c98c/test/e2e/job.go#L83) testcase is falling, need investigate and fix.

**Environment**:
- Kubernetes version (use `kubectl version`):
- Cloud provider or hardware configuration:
- OS (e.g. from /etc/os-release):
- Kernel (e.g. `uname -a`):
- Install tools:
- Others:

### LINKED ISSUE #836: Fix e2e test: statement
**Is this a BUG REPORT or FEATURE REQUEST?**:

/kind bug

**What happened**:
After the revert of batch job patches #806 , the [statement](https://github.com/kubernetes-sigs/kube-batch/blob/5e9977d68a13938fbb6d7dda564da0dca1b1c98c/test/e2e/job.go#L254) testcase is falling, need investigate and fix.

**What you expected to happen**:

**How to reproduce it (as minimally and precisely as possible)**:

**Anything else we need to know?**:

**Environment**:
- Kubernetes version (use `kubectl version`):
- Cloud provider or hardware configuration:
- OS (e.g. from /etc/os-release):
- Kernel (e.g. `uname -a`):
- Install tools:
- Others:

### PR DESCRIPTION
**Pull request by thandayuthapani**
**What this PR does / why we need it**:
- Use KIND instead of DIND
- Add Back Skipped test cases(Gang Scheduling & Statement)
**Which issue(s) this PR fixes** 
Fixes #836 
Fixes #835

### CODE REVIEW

**Comment by thandayuthapani** [comment]
@k82cn Can you re trigger the build...

File: pkg/scheduler/actions/allocate/allocate.go [hunk 1]
```diff
@@ -142,7 +142,8 @@ func (alloc *allocateAction) Execute(ssn *framework.Session) {
 
 			predicateNodes := util.PredicateNodes(task, allNodes, predicateFn)
 			if len(predicateNodes) == 0 {
-				break
+				// When predicates fail for a task in a job, it should go for checking to next task in same job
```
Comment (k82cn) [inline pkg/scheduler/actions/allocate/allocate.go]: please revert this change; if it's necessary, open another PR for it.

File: test/e2e/util.go [hunk 2]
```diff
@@ -518,7 +526,6 @@ func createReplicaSet(context *context, name string, rep int32, img string, req
 					Labels: map[string]string{deploymentName: name},
 				},
 				Spec: v1.PodSpec{
-					SchedulerName: "kube-batch",
```
Comment (k82cn) [inline test/e2e/util.go]: we need this to make sure kube-batch also support RS

**Comment by k82cn** [comment]
CI is still failing :(

**Comment by thandayuthapani** [comment]
@k82cn Fix was raised in this PR https://github.com/kubernetes-sigs/kube-batch/pull/849 
Because of that this PR is failing.

File: test/e2e/job.go [hunk 3]
```diff
@@ -348,7 +346,7 @@ var _ = Describe("Job E2E Test", func() {
 					pri: masterPriority,
 					min: 1,
 					rep: 1,
-					req: twoCPU,
+					req: oneAndHalfCPU,
```
Comment (thandayuthapani) [inline test/e2e/job.go:349]: This change has been made since, max node resource in KIND cluster is 1980m, so 2000m pod cannot be deployed in any of those nodes.  So it has been change to 1500m.

**Comment by thandayuthapani** [comment]
@k82cn CI has been Fixed.
